### PR TITLE
Feature/904 Request Duplicate Check

### PR DESCRIPTION
**Pull request by ggrg**
Code coverage at PR init (to be improved)
```
=============================== Coverage summary ===============================
Statements   : 98.31% ( 3136/3190 )
Branches     : 97.47% ( 770/790 )
Functions    : 98.6% ( 422/428 )
Lines        : 98.29% ( 3100/3154 )
================================================================================

1..1239
# tests 1239
# pass  1239

# ok
```
Please, also note PR https://github.com/mojaloop/central-services-shared/pull/76 needs to be merged in order for this PR to have unit tests and integration tests executed properly.

### CODE REVIEW

**Comment by ggrg** [comment]
```
=============================== Coverage summary ===============================
Statements   : 100% ( 3184/3184 )
Branches     : 99.75% ( 784/786 )
Functions    : 100% ( 427/427 )
Lines        : 100% ( 3148/3148 )
================================================================================

1..1253
# tests 1253
# pass  1253

# ok
```

File: src/models/transfer/facade.js [hunk 1]
```diff
@@ -440,32 +436,73 @@ const saveTransferPrepared = async (payload, stateReason = null, hasPassedValida
...
+          await knex('ilpPacket').transacting(trx).insert(ilpPacketRecord)
+          await knex('transferStateChange').transacting(trx).insert(transferStateChangeRecord)
+          await trx.commit
+        } catch (err) {
+          await trx.rollback
+          throw err
+        }
+      })
+    } else {
+      await knex('transfer').insert(transferRecord)
```
Comment (rmothilal) [inline src/models/transfer/facade.js]: Why would these inserts not be in a transaction? I don't see the reason to only insert certain entries in the database.

File: src/handlers/transfers/handler.js [hunk 2]
```diff
@@ -225,136 +220,127 @@ const fulfil = async (error, messages) => {
...
+          const producer = { functionality: TransferEventType.NOTIFICATION, action: TransferEventAction.COMMIT }
+          return await Util.proceed(params, { consumerCommit, histTimerEnd, fspiopError, producer, fromSwitch })
+        } else if (transfer.transferState !== TransferState.RESERVED) {
+          Logger.info(Util.breadcrumb(location, `callbackErrorNonReservedState--${actionLetter}9`))
+          const fspiopError = ErrorHandler.Factory.createFSPIOPError(ErrorHandler.Enums.FSPIOPErrorCodes.VALIDATION_ERROR, 'non-RESERVED transfer state').toApiErrorObject()
+          const producer = { functionality: TransferEventType.NOTIFICATION, action: TransferEventAction.COMMIT }
+          return await Util.proceed(params, { consumerCommit, histTimerEnd, fspiopError, producer, fromSwitch })
+        } else if (transfer.expirationDate <= new Date()) {
+          Logger.info(Util.breadcrumb(location, `callbackErrorTransferExpired--${actionLetter}10`))
+          // TODO: Previously thrown code was 3300, now - 3303
```
Comment (rmothilal) [inline src/handlers/transfers/handler.js]: Why is this comment here? Do you still need to change it?

File: src/lib/config.js [hunk 3]
```diff
@@ -27,7 +27,8 @@ module.exports = {
   HANDLERS_TIMEOUT_TIMEZONE: RC.HANDLERS.TIMEOUT.TIMEZONE,
   KAFKA_CONFIG: RC.KAFKA,
   PARTICIPANT_INITIAL_POSITION: RC.PARTICIPANT_INITIAL_POSITION,
-  RUN_MIGRATIONS: RC.RUN_MIGRATIONS,
+  RUN_MIGRATIONS: !RC.MIGRATIONS.DISABLED,
```
Comment (rmothilal) [inline src/lib/config.js]: why do you have a ! infront of RC.MIGRATIONS.DISABLED??

File: src/models/transfer/facade.js [hunk 4]
```diff
@@ -33,14 +33,14 @@
...
 const Db = require('../../lib/db')
-const Uuid = require('uuid4')
 const Enum = require('../../lib/enum')
 const TransferExtensionModel = require('./transferExtension')
 const ParticipantFacade = require('../participant/facade')
 const Time = require('../../lib/time')
 const Config = require('../../lib/config')
 const _ = require('lodash')
 const ErrorHandler = require('@mojaloop/central-services-error-handling')
+// TODO: @ggrg Address Logger.info commented out entries below
```
Comment (rmothilal) [inline src/models/transfer/facade.js]: please remove comment

File: src/models/transfer/facade.js [hunk 5]
```diff
@@ -440,32 +436,73 @@ const saveTransferPrepared = async (payload, stateReason = null, hasPassedValida
...
-            return {
-              transferId: payload.transferId,
-              key: ext.key,
-              value: ext.value
-            }
-          })
-          await knex.batchInsert('transferExtension', transferExtensionsRecordList).transacting(trx)
+        await knex('transferParticipant').insert(payerTransferParticipantRecord)
+      } catch (err) {
+        // Logger.info(`Payer transferParticipant insert error: ${err.message}`)
```
Comment (rmothilal) [inline src/models/transfer/facade.js]: Either log or remove

File: src/models/transfer/facade.js [hunk 6]
```diff
@@ -440,32 +436,73 @@ const saveTransferPrepared = async (payload, stateReason = null, hasPassedValida
...
+      }
+      try {
+        await knex('transferParticipant').insert(payeeTransferParticipantRecord)
+      } catch (err) {
+        // Logger.info(`Payee transferParticipant insert error: ${err.message}`)
```
Comment (rmothilal) [inline src/models/transfer/facade.js]: Either log or remove

File: src/models/transfer/facade.js [hunk 7]
```diff
@@ -440,32 +436,73 @@ const saveTransferPrepared = async (payload, stateReason = null, hasPassedValida
...
+          return {
+            transferId: payload.transferId,
+            key: ext.key,
+            value: ext.value
+          }
+        })
+        try {
+          await knex.batchInsert('transferExtension', transferExtensionsRecordList)
+        } catch (err) {
+          // Logger.info(`batchInsert transferExtension error: ${err.message}`)
```
Comment (rmothilal) [inline src/models/transfer/facade.js]: Either log or remove

File: src/models/transfer/facade.js [hunk 8]
```diff
@@ -440,32 +436,73 @@ const saveTransferPrepared = async (payload, stateReason = null, hasPassedValida
...
       } catch (err) {
-        await trx.rollback
-        throw err
+        // Logger.info(`ilpPacket insert error: ${err.message}`)
       }
-    })
+      try {
+        await knex('transferStateChange').insert(transferStateChangeRecord)
+      } catch (err) {
+        // Logger.info(`transferStateChange insert error: ${err.message}`)
```
Comment (rmothilal) [inline src/models/transfer/facade.js]: Either log or remove

File: src/models/transfer/facade.js (hunk 1 above)
Comment (ggrg) [reply src/models/transfer/facade.js]: This is a DA decision - to store what we are able to insert.

File: src/handlers/transfers/handler.js (hunk 2 above)
Comment (ggrg) [reply src/handlers/transfers/handler.js]: Was there before - didn't address it for the current story.

File: src/lib/config.js (hunk 3 above)
Comment (ggrg) [reply src/lib/config.js]: It is negated: RUN_MIGRATION = true when RC.MIGRATIONS.DISABLED is false.

**Pull request merged**